### PR TITLE
#3 LinkedIn single sign-on

### PR DESCRIPTION
**Pull request by rachelchenrc**
- Implemented using OAuth 2.0 based authentication to get LinkedIn user information then create a Firebase Custom Token. 

- Based on the [Firebase official sample code](https://github.com/firebase/functions-samples/tree/master/linkedin-auth)

[Trello card](https://trello.com/c/KMts1No6/3-request-linkedin-single-sign-on)
[Live Env](http://nj-career-network-dev2.firebaseapp.com/)

Screen Shot 2020-02-27 at 4 37 17 PM

### CODE REVIEW

File: functions/service-account.json [hunk 1]
```diff
@@ -0,0 +1,12 @@
+{
```
Comment (gideonred) [inline functions/service-account.json]: What is the purpose of this file?

Comment (rachelchenrc) [reply functions/service-account.json]: provide Service Account credential in the LinkedIn function https://github.com/newjersey/career-network/pull/127/files#diff-1e53ef2f51bd446c876676ba83d7c888

File: pages/_app.js [hunk 2]
```diff
@@ -40,7 +41,11 @@ Sentry.init({
 class MyApp extends App {
   constructor(args) {
     super(args);
+<<<<<<< HEAD
```
Comment (hannahtwobulls) [inline pages/_app.js:44]: Remove!

File: pages/_app.js [hunk 3]
```diff
@@ -77,14 +82,27 @@ class MyApp extends App {
...
     // Remove the server-side injected CSS.
     const jssStyles = document.querySelector('#jss-server-side');
     if (jssStyles && jssStyles.parentNode) {
       jssStyles.parentNode.removeChild(jssStyles);
     }
+<<<<<<< HEAD
 
+<<<<<<< HEAD
     const isIE = !!document.documentMode;
     this.setState({ isIE });
```
Comment (hannahtwobulls) [inline pages/_app.js:100]: Remove!

File: components/Auth/AuthForm.js [hunk 4]
```diff
@@ -42,7 +57,7 @@ export default function AuthForm(props) {
     signInOptions: [
       // Leave the lines as is for the providers you want to offer your users.
       auth.GoogleAuthProvider.PROVIDER_ID,
-      auth.FacebookAuthProvider.PROVIDER_ID,
+      // auth.FacebookAuthProvider.PROVIDER_ID,
```
Comment (rossdakin) [inline components/Auth/AuthForm.js:60]: @gideonred @hannahtwobulls this was code reviewed, yes?

Comment (gideonred) [reply components/Auth/AuthForm.js:60]: This was, yes, but we've clearly missed this important code change. I'm going to look through how myself and others missed this code change.

Comment (rachelchenrc) [reply components/Auth/AuthForm.js:60]: This probably happens during resolving the merge conflict, will double-check on it afterward.

**Pull request merged**